### ISSUE TITLE
Auto disable check/report if respective cluster is unspecified

### ISSUE THREAD

**Issue by rshriram** [istio/proxy#1011]
Since pilot is guaranteed to generate mixer_check_Cluster, mixer_report_cluster, we should auto disable checks/reports if any of these options are missing. This is especially important when user specifies mixer_report_cluster in global config, but does not specify mixer_check_cluster (indicating that she wants to disable check calls). The current way of doing this is a bit too clunky: user has to specify both clusters and then explicitly specify disable_policy_checks. This is error prone and likely to lead to segfaults.

**Comment by qiwzhang** [comment]
Does that mean:  if "check_cluster" in the Transport proto points to a non-existing cluster in the cluster manager,  we should silently disable mixer check.

I think at least we should log an error message for easy debug of such invalid config problem.  But this is per-request error log.  it could flood the log.

**Comment by rshriram** [comment]
No.. if check_cluster is missing or set to empty, we should disable mixer_check and probably log once every 5 minutes or something.

**Comment by qiwzhang** [comment]
Current logic is if check_cluster is empty or not set,  mixer client will set it to default "mixer_server" cluster. 

If we go with your suggestion,  it will be a break change.   The new Pilot should always set "check_cluster" if check_call is needed.    If newer proxy talks to older Pilot,  it will break,  check will not be called.

**Comment by rshriram** [comment]
I have already updated Pilot to always provide check_cluster if its set. Also, we don't have to worry about breaking change now, since global config and such is in alpha mode.
